File: clojure/sublist/src/sublist.clj
(ns sublist)

(defn sublist? [seq1 seq2]
  (some #{seq1} (partition (count seq1) 1 seq2)))

(defn classify [list1 list2]
  (cond
    (= list1 list2) :equal
    (sublist? list1 list2) :sublist
    (sublist? list2 list1) :superlist
    :else :unequal))

File: go/sublist/sublist.go
package sublist

type Relation string

func Sublist(l1, l2 []int) Relation {
	switch {
	case len(l1) == len(l2) && isSublist(l1, l2):
		return "equal"
	case len(l1) > len(l2) && isSublist(l2, l1):
		return "superlist"
	case len(l1) < len(l2) && isSublist(l1, l2):
		return "sublist"
	default:
		return "unequal"
	}
}

func isSublist(l1, l2 []int) bool {
	for j := 0; j < len(l2)-len(l1)+1; j++ {
		match := true
		for i, n := range l1 {
			if n != l2[i+j] {
				match = false
				break
			}
		}
		if match {
			return true
		}
	}
	return false
}
